attest/verifier/config/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+        /// software at this enclave revision.
+        #[serde(default)]
+        mitigated_config_advisories: Vec<String>,
+        /// The list of hardening advisories that are known to be mitigated in
+        /// software at this enclave revision.
+        #[serde(default)]
+        mitigated_hardening_advisories: Vec<String>,
+    },
+}
+
+impl StatusVerifierConfig {
+    /// Build status verifier corresponding to ourself, and add it to a given
+    /// verifier
+    pub fn add_to_verifier(&self, verifier: &mut Verifier) {
+        match self {
+            Self::Mrenclave {
+                MRENCLAVE,
+                mitigated_config_advisories,
+                mitigated_hardening_advisories,
+            } => {
+                let mut mr_enclave_verifier = MrEnclaveVerifier::new(MrEnclave::from(*MRENCLAVE));
+                for advisory in mitigated_config_advisories.iter() {
+                    mr_enclave_verifier.allow_config_advisory(advisory);
+                }
+                for advisory in mitigated_hardening_advisories.iter() {
+                    mr_enclave_verifier.allow_hardening_advisory(advisory);
+                }
+                verifier.mr_enclave(mr_enclave_verifier);
+            }
+            Self::Mrsigner {
+                MRSIGNER,
+                product_id,
+                minimum_svn,
+                mitigated_config_advisories,
+                mitigated_hardening_advisories,
+            } => {
+                let mut mr_signer_verifier =
+                    MrSignerVerifier::new(MrSigner::from(*MRSIGNER), *product_id, *minimum_svn);
+                for advisory in mitigated_config_advisories.iter() {
+                    mr_signer_verifier.allow_config_advisory(advisory);
+                }
+                for advisory in mitigated_hardening_advisories.iter() {
+                    mr_signer_verifier.allow_hardening_advisory(advisory);
+                }
+                verifier.mr_signer(mr_signer_verifier);
+            }
+        }
+    }
+}
+
+/// Defines a json schema for a "trusted-measurements.json" file.
+/// See README.md for example.
+///
+/// The outermost string key of this is the release version number. This is not
+/// interpretted by the software, but could be added as a debug string or
+/// something, and helps with maintenance of the file.
+///
+/// The second string key, within a release, is the name of the enclave that
+/// this is a measurement for.
+///
+/// The VerifierConfig object contains measurement and hardening advisory data
+/// for that enclave at that release.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct TrustedMeasurementSet {
+    table: BTreeMap<String, BTreeMap<String, StatusVerifierConfig>>,
+}
+
+impl TrustedMeasurementSet {
+    /// Create a verifier from this measurement set for a given enclave name.
+    pub fn create_verifier(&self, enclave_name: &str) -> Result<Verifier, Error> {
```

Comment:
in another revision i actually had `&'static str` since I only envision people using this with string constants like `"consensus"`, but I felt like there's no good reason to impose this restriction if it is not required for correctness